02_bubble/vinilapeddireddy/bubblesort_test.go

```diff
@@ -0,0 +1,62 @@
+package main
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+//To test the bubbleFunction with positive numbers
+func TestBubble_PositiveNumbers(t *testing.T) {
+	// Given
+	r := require.New(t)
+	var buf bytes.Buffer
+	out = &buf
+	b := []int{5, 2, 3, 4, 1}
+	actual := bubble(b)
+	expected := []int{1, 2, 3, 4, 5}
+	r.Equalf(expected, actual, "Unexpected output in bubble")
+}
+
+//To test the bubbleFunction negative numbers
+func TestBubble_NegativeNumbers(t *testing.T) {
+	// Given
+	r := require.New(t)
+	var buf bytes.Buffer
+	out = &buf
+	b := []int{-5, 2, 3, 0, 1, 4}
+	actual := bubble(b)
+	expected := []int{-5, 0, 1, 2, 3, 4}
+	r.Equalf(expected, actual, "Unexpected output in bubble")
+}
+
+//To test the copying
+func TestBubble_DoesntModifyInput(t *testing.T) {
+	r := require.New(t)
+	var buf bytes.Buffer
+	out = &buf
+	b := []int{-5, 2, 3, 0, 1, 4}
+	actual := bubble(b)
+
+	// Then
+	r.Equal([]int{-5, 0, 1, 2, 3, 4}, actual)
```

Comment:
This line is not needed in this test. Ideally unit tests will test a single thing and not repeat tests that have been done elsewhere. If a change to your `bubble` function goes bad and the sort order is wrong, but it is still doing the right thing by not mutating the input, this test should continue to pass. It can be a hindrance when many tests start failing because of one change.